### ISSUE TITLE
GPT2ForSequenceClassification accuracy test fails for CUDA AOT Inductor export when its SDPA pattern is mapped to the SDPA op

### ISSUE THREAD

**Issue by sanchitintel** [pytorch/pytorch#122429]
### 🐛 Describe the bug

In #121866, GPT2 SDPA pattern is being mapped to `torch.nn.functional.scaled_dot_product_attention`.
With CUDA, `GPT2ForSequenceClassification` [fails on CI for Inductor AOT export path](https://github.com/pytorch/pytorch/actions/runs/8319111885/job/22762567770), but not in other trunk CI checks.

Disabling this pattern's replacement for CUDA, until this issue would be resolved.
For attaining better performance, this issue should be fixed.

Thanks!

### Versions

PyTorch main branch

cc @ezyang @msaroufim @bdhirsh @anijain2305 @zou3519 @chauhang @desertfire @chenyang78 @drisspg @eellison

**Comment by jansel** [comment]
Can you explain a bit more what the issue is?  Why is this AOT Inductor specific?

**Comment by sanchitintel** [comment]
Sorry, I added that tag because the only CI check that fails when that pattern's replacement is enabled for CUDA is `cuda12.1-py3.10-gcc9-sm86 / test (aot_inductor_huggingface)`.

**Comment by jansel** [comment]
cc @eellison @desertfire

**Comment by IvanKobzarev** [comment]
Hi, @desertfire , I am doing issue scrapping. I see you have a landed PR for this issue, is it fixed and issue can be closed?

**Comment by desertfire** [comment]
> Hi, @desertfire , I am doing issue scrapping. I see you have a landed PR for this issue, is it fixed and issue can be closed?

Haven't get a chance to look into it. Created https://github.com/pytorch/pytorch/pull/137085 to see if the problem still exists.
